lib/dom.js

```diff
@@ -53,6 +53,40 @@ declare class FileList {
     [index: number]: File;
 }
 
+/* DataTransfer */
+
+declare class DataTransfer {
+  dropEffect: string;
+  effectAllowed: string;
+
+  items: DataTransferItemList; // readonly
+
+  setDragImage(image: Element, x: number, y: number);
+
+  /* old interface */
+  types: array<string>; // readonly
+  getData(format: string): string;
+  setData(format: string, data: string);
+  clearData(format: ?string);
+  files: FileList; // readonly
+}
+
+declare class DataTransferItemList {
+  length: number; // readonly
+  [index: number]: DataTransferItem;
+  add(data: string, type: string): ?DataTransferItem;
+  add(data: File): ?DataTransferItem;
+  remove(index: number);
+  clear();
+};
+
+declare class DataTransferItem {
+  kind: string; // readonly
+  type: string; // readonly
+  getAsString(_callback: ?(data: string) => void);
```

Comment:
I've been typing the return type of callbacks as `mixed` recently. People really like to use arrow functions as callbacks, like

```JavaScript
  item.getAsString(data => foo(data))
```

and if `foo()` doesn't return `void` then you would get a type error.